src/main/groovy/pl/allegro/tech/build/axion/release/domain/NextVersionMarker.groovy

```diff
@@ -9,22 +10,34 @@ class NextVersionMarker {
 
     private static final ReleaseLogger logger = ReleaseLogger.Factory.logger(NextVersionMarker)
 
-    private final ScmService repository
+    private final ScmService repositoryService
+    private final VersionConfig versionConfig
 
-    NextVersionMarker(ScmService repository) {
-        this.repository = repository
+    NextVersionMarker(ScmService repositoryService, VersionConfig config) {
```

Comment:
I try not to inject `VersionConfig` object, as it is very gradle-specific. Instead you could pass version to `markNextVersion` method. You have access to `VersionConfig` in `MarkNextVersionTask`, from where this method is called.